### PR TITLE
gh-121351: skip test_not_wiping_history_file() if no readline

### PR DESCRIPTION
**Pull request by skirpichev**
* Issue: gh-121351

### CODE REVIEW

**Comment by skirpichev** [comment]
Can skip news, I think.

Should be tested with buildbots, sorry:(

File: Lib/test/test_pyrepl/test_pyrepl.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
     more_lines,
     multiline_input,
     code_to_events,
```
Comment (vstinner) [inline Lib/test/test_pyrepl/test_pyrepl.py]: The import is wrong.

**Comment by vstinner** [comment]
!buildbot AMD64 FreeBSD14

**Comment by vstinner** [comment]
!buildbot ARM64 MacOS M1 NoGIL

**Comment by vstinner** [comment]
!buildbot s390x Fedora PR

**Review by vstinner — APPROVED** [review]
LGTM. But let's see if it does fix the issue on buildbots.

**Comment by smontanaro** [comment]
I still see this failure on my Mac. I changed the failing test to skip it if `readline.backend == "editline"`. I'm not sure that's the right change, but it would seem that `editline` and the real `readline` must differ in some way related to this.

```
diff --git a/Lib/test/test_pyrepl/test_pyrepl.py b/Lib/test/test_pyrepl/test_pyrepl.py
index 015b6905662..0f055edfcc8 100644
--- a/Lib/test/test_pyrepl/test_pyrepl.py
+++ b/Lib/test/test_pyrepl/test_pyrepl.py
@@ -904,8 +904,10 @@ def test_python_basic_repl(self):
 
     def test_not_wiping_history_file(self):
         # skip, if readline module is not available
-        import_module('readline')
+        readline = import_module('readline')
 
+        if readline.backend == "editline":
+            self.skipTest("editline's history mgmt isn't 100% compatible w/readline")
         hfile = tempfile.NamedTemporaryFile(delete=False)
         self.addCleanup(unlink, hfile.name)
         env = os.environ.copy()
```

If you think that is the right thing to do, I will create a PR, but I'm kind of busy with some other stuff right now. Might take a couple days to get around to it.

**Comment by skirpichev** [comment]
> I still see this failure on my Mac.

Here is the recent test on main: https://buildbot.python.org/#/builders/1270/builds/2021  So, I doubt it's a good idea just disable this on Mac.

@smontanaro, could you, please, test original issue #121245?  I worry it might be valid on your system.

Is it possible that you have some PYTHON* variables defined?

**Comment by smontanaro** [comment]
Rebuilding the `readline` module to use the readline 8.2.10 library in `/opt/homebrew`, I still get the same failure. 

Note, however, that if I unset `PYTHONSTARTUP`, with the current code, the failing test passes. I believe my personal readline setup is interfering. @vstinner modified another readline test to run in isolated mode (see #121414). It would seem the same change needs to be applied here.

**Comment by vstinner** [comment]
> Note, however, that if I unset PYTHONSTARTUP, with the current code, the failing test passes.

See https://github.com/python/cpython/pull/121672 in this case.

**Pull request merged**